bindings/muen/muen-net.c

```diff
@@ -71,16 +92,38 @@ solo5_result_t solo5_net_read(uint8_t *buf, size_t size, size_t *read_size)
     }
 }
 
-bool muen_net_pending_data()
+solo5_result_t solo5_net_acquire(const char *name, solo5_handle_t *h,
+        struct solo5_net_info *info)
 {
-    return muen_channel_has_pending_data(net_in, &net_rdr);
+    solo5_handle_t handle = 0;
+    unsigned mft_index;
+    struct mft_entry *mft_e = mft_get_by_name(muen_manifest, name,
+            MFT_DEV_NET_BASIC, &mft_index);
+    if (mft_e == NULL)
+        return SOLO5_R_EINVAL;
+
+    handle = (solo5_handle_t)mft_index;
+
+    if (devices[handle].acquired) {
+        log(WARN, "Solo5: Net: Device '%s' already acquired\n", name);
+        return SOLO5_R_EINVAL;
+    }
```

Comment:
I see what you're doing here -- `solo5_net_acquire()` does the actual initialisation by calling `muen_net_dev_init()`, and errors if it's called more than once. This is slightly out of spec, in that I had intended the `_acquire_` APIs to allow for multiple calls.

Having said that, I can see how `_acquire_` implies a transition of some kind, so perhaps it'd be better to just change the docs (solo5.h) to say that you're only allowed to call it once per handle?

Reply:
Ok, I'll formalise that in a separate PR.